distributed/worker.py

```diff
@@ -3077,7 +3123,7 @@ async def gather_dep(
                 for k in self.in_flight_workers[worker]:
                     ts = self.tasks[k]
                     recommendations[ts] = tuple(msg.values())
-                raise
+                return GatherDepDoneEvent(stimulus_id=stimulus_id)
```

Comment:
The exception is now shielded from ``with self.log_errors():``. Previously it was double reported as it is already logged by ``logger.exceptions(e)`` on line 3117.